prow/config/inrepoconfig_test.go

```diff
@@ -631,3 +639,80 @@ func testDefaultProwYAMLGetter_RejectsNonGitHubRepo(clients localgit.Clients, t
 		t.Errorf("Error %v does not have expected message %s", err, expectedErrMsg)
 	}
 }
+
+type testClientFactory struct {
+	git.ClientFactory // This will be nil during testing, we override the functions that are used.
+	clientsCreated    int
+}
+
+func (cf *testClientFactory) ClientFor(org, repo string) (git.RepoClient, error) {
+	cf.clientsCreated++
+	// Returning this RepoClient ensures that only Fetch() is called and that Close() is not.
+	return &fetchOnlyNoCleanRepoClient{}, nil
+}
+
+type fetchOnlyNoCleanRepoClient struct {
+	git.RepoClient // This will be nil during testing, we override the functions that are allowed to be used.
+}
+
+func (rc *fetchOnlyNoCleanRepoClient) Fetch() error {
+	return nil
+}
+
+// TestInRepoConfigGitCacheConcurrency validates the following properties of InRepoConfigGitCache
+// - RepoClients are protected from concurrent use.
+// - Use of a RepoClient for one repo does not prevent concurrent use of a RepoClient for another repo.
+// - RepoClients are reused, only 1 client is created per repo.
+func TestInRepoConfigGitCacheConcurrency(t *testing.T) {
+	t.Parallel()
+
+	cf := &testClientFactory{}
+	cache := NewInRepoConfigGitCache(cf)
+	org, repo1, repo2 := "org", "repo1", "repo2"
+	// block channels are populated from the main thread, signal channels are read from the main thread.
+	signal := make(chan bool)
+	sharedRepo1State := 0
```

Comment:
Delete?

Reply:
Both the thread1 and thread2 routine write to it, which will make the race detector go off if `ClientFor` allows them to both have a client